src/backend/distributed/planner/merge_planner.c

```diff
@@ -442,23 +422,40 @@ ErrorIfMergeHasUnsupportedTables(Query *parse, List *rangeTableList,
 			continue;
 		}
 
+		if (IsCitusTableType(relationId, REFERENCE_TABLE))
```

Comment:
I guess making it `if  - else if - else if` or a `switch` could be easier to follow && more robust (as we might forget to `continue` etc)

Reply:
Completely refactored it, please take a look